### PR TITLE
[v14] feat: Disable auto-enroll via environment variable

### PR DESCRIPTION
**Pull request by codingllama**
Backport #47679 and #47723 to branch/v14

changelog: Auto-enroll may be locally disabled using the `TELEPORT_DEVICE_AUTO_ENROLL_DISABLED=1` environment variable

### CODE REVIEW

**Comment by codingllama** [comment]
This is flaky! Mailed #47723, will merge it here once it lands.

- #47723

**Review by ryanclark — APPROVED** [review]

**Comment by codingllama** [comment]
PTAL @rosstimothy.

**Review by rosstimothy — APPROVED** [review]

**Pull request merged**
